CMakeLists.txt

```diff
@@ -39,6 +39,15 @@ else()
 endif()
 # ------------------
 
+# ------------------
+# Extensive mode for running extension one API tests
+option(SYCL_CTS_ENABLE_EXTENSION_ONE_API_TESTS
+       "Enable extension one API tests" OFF)
```

Comment:
I suppose there will be more than one "oneAPI" extension and some implementation might want to validate some, but not all of them. We probably should think about more flexible way to configure tests for extensions (e.g. re-use filters enabled for core feature tests).

In addition to that, it would be great to build a common infrastructure for adding tests for other vendor extensions: `intel`, `codeplay`, etc.

Reply:
> In addition to that, it would be great to build a common infrastructure for adding tests for other vendor extensions: intel, codeplay, etc.

@bader could you clarify what this infrastructure should do?
it should be a common flag to enable to make possible to build all extensions tests from all vendors or something different?

Reply:
Give developers control over enabled tests. This switch provides just two options: enable or disable all oneAPI extension tests at once.
I think we should have an interface to enable or disable each particular extension test.

Reply:
> I think we should have an interface to enable or disable each particular extension test.

@bader ok, `SYCL_CTS_ENABLE_EXTENSION_ONE_API_TESTS` will be switched into separate flag for each extension group

Reply:
the flag `SYCL_CTS_ENABLE_EXTENSION_ONE_API_TESTS` was replaced with `SYCL_EXT_ONEAPI_ENABLE_SUB_GROUP_MASK_TESTS` and new flag related to the tests on `sub_group_mask`
I've done this change [here](https://github.com/KhronosGroup/SYCL-CTS/pull/200/commits/e2188e3ff5458af4ef8a8f6bb2d9447088bc8846)